### PR TITLE
Vault initial deposit to `0xDeaD`

### PR DESCRIPTION
**Pull request by Rubilmax**
# Pull Request

## Issue(s) fixed

This pull request:

- fixes https://github.com/spearbit-audits/review-morpho-june/issues/25

### CODE REVIEW

File: src/extensions/SupplyVault.sol [hunk 1]
```diff
@@ -69,7 +69,7 @@ contract SupplyVault is ISupplyVault, ERC4626UpgradeableSafe, Ownable2StepUpgrad
         __Ownable_init_unchained(); // Equivalent to __Ownable2Step_init
         __ERC20_init_unchained(name, symbol);
         __ERC4626_init_unchained(ERC20Upgradeable(newUnderlying));
-        __ERC4626UpgradeableSafe_init_unchained(initialDeposit);
+        __ERC4626UpgradeableSafe_init_unchained(initialDeposit, address(0));
```
Comment (pakim249CAL) [inline src/extensions/SupplyVault.sol:72]: I think it's better to use a designated burn address than the zero address, so we don't have to think about the consequences of using the zero address.

Comment (Rubilmax) [reply src/extensions/SupplyVault.sol:72]: Do you have a concrete example of consequences of using address zero? I genuinely don't see what you mean

Comment (pakim249CAL) [reply src/extensions/SupplyVault.sol:72]: I actually don't think there are any consequences in this particular case, and believe it to be safe. My concern is more that the wider smart contract ecosystem seems to treat the zero address as something that is invalid, thus creating a vector of uncertainty among readers and integrators of the code. If an obvious burn address is used like 0xdead, then it becomes more obvious to any reader of the code that the initial deposit is meant to be burned.

**Comment by MerlinEgalite** [comment]
why is it still a draft?

**Review by Tristan22400 — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by makcandrov — APPROVED** [review]

**Review by pakim249CAL — APPROVED** [review]

**Pull request merged**